src/components/Footer/Footer.tsx

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+import { FilterType } from '../../types/FilterType';
+import { deleteTodo } from '../../api/todos';
+import { TextError } from '../../types/TextError';
+
+type Props = {
+  typeOfFilter: string;
+  setTypeOfFilter: (filter: string) => void;
+  todos: Todo[];
+  setTodos: (todos: Todo[]) => void;
+  setError: (value: TextError) => void;
+  completedTodoList: Todo[];
+};
+
+export const Footer: React.FC<Props> = ({
+  typeOfFilter,
+  setTypeOfFilter,
+  todos,
+  setTodos,
+  setError,
+  completedTodoList,
+}) => {
+  const activeTodos = todos.filter(todo => !todo.completed);
+
+  const deleteFinishedTodos = async (finished: Todo[]) => {
+    try {
+      finished.forEach(async (todo) => {
+        await deleteTodo(todo.id);
+
+        setTodos([...activeTodos]);
+      });
+    } catch {
+      setError(TextError.Delete);
+    }
+  };
+
+  return (
+    <footer className="todoapp__footer" data-cy="Footer">
+      <span className="todo-count" data-cy="todosCounter">
+        {`${activeTodos.length} items left`}
+      </span>
+
+      <nav className="filter" data-cy="Filter">
+        <a
+          data-cy="FilterLinkAll"
+          href="#/"
+          className={classNames(
+            'filter__link',
+            { selected: typeOfFilter === FilterType.All },
+          )}
+          onClick={() => setTypeOfFilter(FilterType.All)}
+        >
+          All
+        </a>
+
+        <a
+          data-cy="FilterLinkActive"
+          href="#/active"
+          className={classNames(
+            'filter__link',
+            { selected: typeOfFilter === FilterType.Active },
+          )}
+          onClick={() => setTypeOfFilter(FilterType.Active)}
+        >
+          Active
+        </a>
+        <a
+          data-cy="FilterLinkCompleted"
+          href="#/completed"
+          className={classNames(
+            'filter__link',
+            { selected: typeOfFilter === FilterType.Completed },
+          )}
+          onClick={() => setTypeOfFilter(FilterType.Completed)}
+        >
+          Completed
+        </a>
+      </nav>
+
+      <button
+        data-cy="ClearCompletedButton"
+        type="button"
+        className="todoapp__clear-completed"
+        onClick={() => deleteFinishedTodos(completedTodoList)}
+      >
+        {completedTodoList.length > 0 && 'Clear completed'}
```

Comment:
Check how this button works on [the working example](https://mate-academy.github.io/react_todo-app-with-api/) and yours.
<img width="904" alt="Screenshot 2022-10-10 at 19 59 08" src="https://user-images.githubusercontent.com/17098736/194918273-358997cc-2859-4833-a3a8-6eb779931958.png">